lib/ansible/config/base.yml

```
@@ -791,18 +791,22 @@ DEFAULT_HASH_BEHAVIOUR:
     - This setting controls how variables merge in Ansible.
       By default Ansible will override variables in specific precedence orders, as described in Variables.
       When a variable of higher precedence wins, it will replace the other value.
-    - "Some users prefer that variables that are hashes (aka 'dictionaries' in Python terms) are merged.
-      This setting is called 'merge'. This is not the default behavior and it does not affect variables whose values are scalars
-      (integers, strings) or arrays.  We generally recommend not using this setting unless you think you have an absolute need for it,
-      and playbooks in the official examples repos do not use this setting"
+    - Setting this option to 'merge' combines common top-level keys in hashes (aka 'dictionaries' in Python terms) instead
+      of replacing them. It does not affect variables whose values are scalars (integers, strings) or arrays.
+      This setting is strongly discouraged and indicates variables should be restructured. You can use the ``combine`` filter
+      to merge hashes transparently.
+      This setting may be used for a specific objective (for example, merging variables from inventory sources), but because
```

Making this a new paragraph in the output might make it more readable (same for further below):
```suggestion
    - This setting may be used for a specific objective (for example, merging variables from inventory sources), but because
```